t/synopsis.pl

```diff
@@ -4,3 +4,11 @@
 synopsis :-
     '99 Bottles of Beer' =~ '[0-9]+ bottles'/i,
     writeln('Take one down...').
+
+'works with pattern in variable' :-
+    Pattern = '[a-z]',
+    'a' =~ Pattern.
+
+'works with text in variable' :-
+    Text = a,
+    Text =~ '[a-z]'.
```

Comment:
These tests look good.  The file t/synopsis.pl is for testing the code that's in the [Synopsis section](https://github.com/mndrix/regex/tree/c264f36a5cbbac20c1acafe7b770e7fe93334a7b#synopsis) of the readme file, so these tests should probably be in t/examples.pl or maybe a new file t/regressions.pl